core/dbt/helper_types.py

```diff
@@ -80,12 +80,15 @@ class NVEnum(StrEnum):
     def __eq__(self, other):
         return isinstance(other, NVEnum)
 
+    def __call__(self):
+        return self.novalue
+
 
 @dataclass
 class NoValue(dbtClassMixin):
     """Sometimes, you want a way to say none that isn't None"""
 
-    novalue: NVEnum = NVEnum.novalue
+    novalue: NVEnum = field(default_factory=NVEnum.novalue)
```

Comment:
The approach in #6321 is to instead do:
```suggestion
    novalue: NVEnum = field(default_factory=lambda: NVEnum.novalue)
```
I think that avoids the need to define the `__call__` method above

Reply:
I think the approach in #6321 is preferable for a couple reasons:
1. It's more consistent with how other `field.default_factory` values are set throughout the codebase, and 
2. It's more locally contained and easier to reason about - no need to create/rely on a `__call__` method on another class to get `NoValue` working

Reply:
Thought so!